### DISCUSSION TITLE
Generating raw SQL queries from a GraphQL query

### DISCUSSION THREAD

**Discussion by kangshung** [RobertCraigie/prisma-client-py#410] [Q&A]
Hi, is there any way to use your query engine in a way that it returns specific raw SQL queries (for specific databases) provided it gets a GraphQL query and schema as input? If so, I would be really grateful if you could enlighten me how to do it.

**Comment by RobertCraigie** [comment]
It is currently not possible to extract the raw SQL queries that are generated unfortunately.

You can set the `log_queries` option on the `Prisma` instance which will then log everything that the query engine outputs. This feature is undocumented as there are some rough edges:

- The logs are very noisy and contain information unrelated to the generated queries
- The logged SQL queries are not pretty. IIRC they may contain raw newline characters

Nonetheless you can try it out like this:

```py
prisma = Prisma(log_queries=True)
```

Refer to #31 for tracking this feature.

**Reply by kangshung** [reply]
Thanks for the quick response!
